app/workers/classification_count_worker.rb

```diff
@@ -3,16 +3,19 @@ class ClassificationCountWorker
 
   sidekiq_options queue: :data_medium
 
-  def perform(subject_id, workflow_id)
+  def perform(subject_id, workflow_id, was_update=false)
     workflow = Workflow.find(workflow_id)
 
     if workflow.project.live
-      count = SubjectWorkflowStatus.find_or_create_by!(subject_id: subject_id, workflow_id: workflow_id)
+      count = nil
 
-      count.class.increment_counter(:classifications_count, count.id)
-      SubjectWorkflowStatusCountWorker.perform_async(count.id)
+      Workflow.transaction do
+        count = SubjectWorkflowStatus.find_or_create_by!(subject_id: subject_id, workflow_id: workflow_id)
+        count.class.increment_counter(:classifications_count, count.id) unless was_update
+      end
 
       ProjectClassificationsCountWorker.perform_async(workflow.project.id)
+      SubjectWorkflowStatusCountWorker.perform_async(count.id)
```

Comment:
If `find_or_create_by!` raises, AR's `.transaction` catches that exception to perform the rollback, but it also reraises so that the exception itself bubbles up. So in case of exception, this line will never be reached.